### PR TITLE
CONTROLLERS: GetById Post

### PR DESCRIPTION
**Pull request by cacti79**

### CODE REVIEW

**Comment by ShriHumrudha** [comment]
> _No description provided._

Please add screenshots for the different scenarios - 200, 400, 404.

**Comment by cacti79** [comment]
image
image
image

File: Taarafo.Core/Controllers/PostsController.cs [hunk 1]
```diff
@@ -72,6 +72,34 @@ public ActionResult<IQueryable<Post>> GetAllPosts()
...
+        public async ValueTask<ActionResult<Post>> GetPostByIdAsync(Guid postId)
+        {
+            try
+            {
+                Post post = await this.postService.RetrievePostByIdAsync(postId);
+
+                return Ok(post); 
+            }
+            catch(PostValidationException postValidationException)
+                when(postValidationException.InnerException is NotFoundPostException )
```
Comment (hassanhabib) [inline Taarafo.Core/Controllers/PostsController.cs:85]: Run clean up son. @cacti79

File: Taarafo.Core/Controllers/PostsController.cs [hunk 2]
```diff
@@ -108,6 +136,6 @@ public async ValueTask<ActionResult<Post>> DeletePostByIdAsync(Guid postId)
             {
                 return InternalServerError(postServiceException);
             }
-        }
+        }
```
Comment (hassanhabib) [inline Taarafo.Core/Controllers/PostsController.cs:139]: Why do we need this extra space here?

**Pull request merged**